SimExLite/PMICalculators/SimpleScatteringPMICalculator.py

```diff
@@ -0,0 +1,11147 @@
+        cc = cc + 1
+    return all_disp
+
+
+def s2e_gen_randrot_quat(quat, rotmat):
+
+    if (
+        0
+        == quat[0] * quat[0] + quat[1] * quat[1] + quat[2] * quat[2] + quat[3] * quat[3]
+    ):
+        u0, u1, u2 = numpy.random.random(3)
+        q0 = numpy.sqrt(1.0 - u0) * numpy.sin(2 * numpy.pi * u1)
+        q1 = numpy.sqrt(1.0 - u0) * numpy.cos(2 * numpy.pi * u1)
+        q2 = numpy.sqrt(u0) * numpy.sin(2 * numpy.pi * u2)
+        q3 = numpy.sqrt(u0) * numpy.cos(2 * numpy.pi * u2)
+
+        quat[0] = q0
+        quat[1] = q1
+        quat[2] = q2
+        quat[3] = q3
+
+    else:
+        q0 = quat[0]
+        q1 = quat[1]
+        q2 = quat[2]
+        q3 = quat[3]
+
+    rotmat[0] = q0 * q0 + q1 * q1 - q2 * q2 - q3 * q3
+    # // M[0][0]
+    rotmat[1] = 2 * (q1 * q2 - q0 * q3)
+    # // M[0][1]
+    rotmat[2] = 2 * (q1 * q3 + q0 * q2)
+    # // M[0][2]
+    rotmat[3] = 2 * (q1 * q2 + q0 * q3)
+    # // M[1][0]
+    rotmat[4] = q0 * q0 - q1 * q1 + q2 * q2 - q3 * q3
+    # // M[1][1]
+    rotmat[5] = 2 * (q2 * q3 - q0 * q1)
+    # // M[1][2]
+    rotmat[6] = 2 * (q1 * q3 - q0 * q2)
+    # // M[2][0]
+    rotmat[7] = 2 * (q2 * q3 + q0 * q1)
+    # // M[2][1]
+    rotmat[8] = q0 * q0 - q1 * q1 - q2 * q2 + q3 * q3
+    # // M[2][2]
+
+
+#        inversematrix3( (void*) (rotmat + ii) , (void*) (invmat + ii) ) ;
+
+#    for ( ii = 0 ; ii < randrot ; ii++ )
+#    {
+#        for ( jj=0 ; jj < 3 ; jj++ )
+#        {
+#            printf( "% e % e % e\n" , rotmat[ii][jj*3+0] ,  rotmat[ii][jj*3+1] ,
+#                                      rotmat[ii][jj*3+2] ) ;
+#        }
+#        printf("\n") ;
+
+#        for ( jj=0 ; jj < 3 ; jj++ )
+#        {
+#            printf( "% e % e % e\n" , invmat[ii][jj*3+0] ,  invmat[ii][jj*3+1] ,
+#                                      invmat[ii][jj*3+2] ) ;
+#        }
+#        printf("\n\n\n") ;
+#    }
+# }
+
+
+##############################################################################
+
+
+def s2e_rand_orient(r, mat):
+
+    N = r.shape[0]
+    ###    print N
+    vv = numpy.zeros((3, 0))
+    for ii in range(N):
+        vv = r[ii, :]
+        ###        print vv , vv.shape , mat.shape
+        r[ii, 0] = mat[0] * vv[0] + mat[1] * vv[1] + mat[2] * vv[2]
+        r[ii, 1] = mat[3] * vv[0] + mat[4] * vv[1] + mat[5] * vv[2]
+        r[ii, 2] = mat[6] * vv[0] + mat[7] * vv[1] + mat[8] * vv[2]
+
+
+##############################################################################
+def f_eval_numE(a_snp, a_sample):
+
+    num_Z = len(list(a_sample["selZ"].keys()))
+    all_numE = numpy.zeros((num_Z,))
+    cc = 0
+    for sel_Z in list(a_sample["selZ"].keys()):
+        all_numE[cc] = numpy.mean(a_snp["q"][a_sample["selZ"][sel_Z]])
+        cc = cc + 1
+    return all_numE
+
+
+##############################################################################
+
+
+def f_md_step(r, v, m, dt):
+    r = r.copy()
+    v = v.copy()
+    a = (force(config, param) + external_force(config, param)) / m
```

Comment:
`config` and `param` are undefined, so this will crash.

Reply:
Function not used.